processor/processor.go

```diff
@@ -0,0 +1,194 @@
+package processor
+
+import (
+	"fmt"
+	"path"
+
+	"github.com/algorand/go-algorand/agreement"
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/rpcs"
+	"github.com/algorand/indexer/accounting"
+)
+
+type Processor interface {
+	Process(cert *rpcs.EncodedBlockCert) error
+	SetHandler(handler func(block *ledgercore.ValidatedBlock) error)
+	GetLastProcessedRound() uint64
```

Comment:
I guess I propogated that initialization error by suggesting this interface. It should be much easier by changing to this format.
```suggestion
	NextRoundToProcess() uint64
```